tree/tree/inc/TBasket.h

```diff
@@ -54,20 +54,25 @@ class TBasket : public TKey {
-   Int_t       fNevBuf{0};                    ///< Number of entries in basket
-   Int_t       fLast{0};                      ///< Pointer to last used byte in basket
-   Bool_t      fHeaderOnly{kFALSE};           ///< True when only the basket header must be read/written
-   UChar_t     fIOBits{0};                    ///<!IO feature flags.  Serialized in custom portion of streamer to avoid forward compat issues unless needed.
-   Bool_t      fOwnsCompressedBuffer{kFALSE}; ///<! Whether or not we own the compressed buffer.
-   Bool_t      fReadEntryOffset{kFALSE};      ///<!Set to true if offset array was read from a file.
-   Int_t      *fDisplacement{nullptr};        ///<![fNevBuf] Displacement of entries in fBuffer(TKey)
-   Int_t      *fEntryOffset{nullptr};         ///<[fNevBuf] Offset of entries in fBuffer(TKey); generated at runtime.  Special value
-                                              /// of `-1` indicates that the offset generation MUST be performed on first read.
-   TBranch    *fBranch{nullptr};              ///<Pointer to the basket support branch
-   TBuffer    *fCompressedBufferRef{nullptr}; ///<! Compressed buffer.
-   Int_t       fLastWriteBufferSize{0};       ///<! Size of the buffer last time we wrote it to disk
+   Int_t       fBufferSize{0};                    ///< fBuffer length in bytes
+   Int_t       fNevBufSize{0};                    ///< Length in Int_t of fEntryOffset OR fixed length of each entry if fEntryOffset is null!
+   Int_t       fNevBuf{0};                        ///< Number of entries in basket
+   Int_t       fLast{0};                          ///< Pointer to last used byte in basket
+   Bool_t      fHeaderOnly{kFALSE};               ///< True when only the basket header must be read/written
+   UChar_t     fIOBits{0};                        ///<!IO feature flags.  Serialized in custom portion of streamer to avoid forward compat issues unless needed.
+   Bool_t      fOwnsCompressedBuffer{kFALSE};     ///<! Whether or not we own the compressed buffer.
+   Bool_t      fReadEntryOffset{kFALSE};          ///<!Set to true if offset array was read from a file.
+   Int_t      *fDisplacement{nullptr};            ///<![fNevBuf] Displacement of entries in fBuffer(TKey)
+   Int_t      *fEntryOffset{nullptr};             ///<[fNevBuf] Offset of entries in fBuffer(TKey); generated at runtime.  Special value
+                                                  /// of `-1` indicates that the offset generation MUST be performed on first read.
+   TBranch    *fBranch{nullptr};                  ///<Pointer to the basket support branch
+   TBuffer    *fCompressedBufferRef{nullptr};     ///<! Compressed buffer.
+   Int_t       fLastWriteBufferSize[3] = {0,0,0}; ///<! Size of the buffer last three buffers we wrote it to disk
+#ifdef R__TRACK_BASKET_ALLOC_TIME
+   ULong64_t   fResetAllocationTime{0};           ///<! Time spent reallocating baskets in microseconds during last Reset operation.
```

Comment:
since the previous data member is a int[3], I think the order (bool : 1, short, ULong64_t) would result is a (slightly) smaller over sizeof.